akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceWithContextSpec.scala

```diff
@@ -28,6 +28,16 @@ class SourceWithContextSpec extends StreamSpec {
         .expectComplete()
     }
 
+    "get created from a source of tuple2" in {
+      val msg = Message("a", 1L)
+      SourceWithContext.fromPairs(Source(Vector((msg, msg.offset))))
```

Comment:
in scaladsl it might be more idiomatic to name it `fromTuples`?

in javadsl it's good with `fromPairs`